Algorithms/Balanced_Parenthesis.cpp

```diff
@@ -0,0 +1,62 @@
+#include <iostream>
+#include <stack>
+#include <string>
+using namespace std;
+bool check(char x, char y)
+{
+	if ((x=='(' && y==')') || (x=='{' && y=='}') || (x=='[' && y==']') || (x=='<' && y=='>'))
+	{
+		return true;
+	}
+	else
+	{
+		return false;
+	}
+}
+
+bool checkbalancedparenthesis(string s)
+{
+	stack<char> st;
+	int top=-1;
```

Comment:
Add appropriate comments to make it clear why you have initialized **top** with -1.

Also, add comments everywhere necessary to make things clear.